src/main/java/pl/byczazagroda/trackexpensesappbackend/exception/WalletExceptionHandler.java

```diff
@@ -0,0 +1,18 @@
+package pl.byczazagroda.trackexpensesappbackend.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class WalletExceptionHandler {
+
+    @ExceptionHandler(WalletNotSavedException.class)
+    public ResponseEntity<AppRuntimeException> walletNotSavedHandler(WalletNotSavedException exception) {
+
+        HttpStatus httpStatus = HttpStatus.INTERNAL_SERVER_ERROR;
+
+        return new ResponseEntity<>(exception, httpStatus);
```

Comment:
**Should be:**
return new ResponseEntity<>(exception ...   // ?
**or**
return new ResponseEntity<>(exception.getMessage() ...   // ?

**In other words:**
What for exception object on fornt-end side instead of only exception text message of it.
To be honest I'm not sure what will execute in this ResponseEntity constructor - exception going to be message header or objet as body.

[ResponseEntity](https://docs.spring.io/spring-framework/docs/3.0.x/javadoc-api/org/springframework/http/ResponseEntity.html#ResponseEntity(T,%20org.springframework.http.HttpStatus))([T](https://docs.spring.io/spring-framework/docs/3.0.x/javadoc-api/org/springframework/http/ResponseEntity.html) body, [HttpStatus](https://docs.spring.io/spring-framework/docs/3.0.x/javadoc-api/org/springframework/http/HttpStatus.html) statusCode)
          Create a new ResponseEntity with the given body and status code, and no headers.

So looks like it's going as object. The question what will be serialized from exception object. We need there only exception message - not other information (ex.: Stack Trace if it exist there [**1**]).

[Get Stack Trace from exception](https://www.baeldung.com/java-stacktrace-to-string)

But still what will be extracted in this case? `e.toString()` ? And what this will contain there? Only message of exception or other informations?

Reply:
You can define HttpStatus as an annotation, as in the example below.
https://github.com/Simple-as-Coding/tutoring-platform/blob/main/src/main/java/pl/simpleascoding/tutoringplatform/exception/UserIsAlreadyATeacherException.java